services/server/server.go

```diff
@@ -731,19 +736,25 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
 		}
 	}
 
+	_, err = models.GetPerconaSSODetails(s.db.Querier)
+
 	return &serverpb.ChangeSettingsResponse{
-		Settings: s.convertSettings(newSettings),
+		Settings: s.convertSettings(newSettings, err == nil),
 	}, nil
 }
 
 // UpdateConfigurations updates supervisor config and requests configuration update for VictoriaMetrics components.
 func (s *Server) UpdateConfigurations() error {
 	settings, err := models.GetSettings(s.db)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "failed to get settings")
 	}
-	if err := s.supervisord.UpdateConfiguration(settings); err != nil {
-		return err
+	ssoDetails, err := models.GetPerconaSSODetails(s.db.Querier)
+	if err != nil {
+		s.l.Warnf("Failed to get Percona SSO Details: %v. Percona SSO is most likely not set up.", err)
```

Comment:
can we get an exact reason but not possible? Are there any other possible reasons ?

Reply:
I can check for exact error when the percona sso is not set up and not log anything. If the error is any other error, like database connection failure, I return error as in the case of getting settings on lines above.

Reply:
done